src/io/csv/type_conversion.cuh

```diff
@@ -0,0 +1,299 @@
+/* Copyright 2017 NVIDIA Corporation.  All rights reserved. */
+
+#ifndef CONVERSION_FUNCTIONS_CUH
+#define CONVERSION_FUNCTIONS_CUH
+
+#include <math_functions.h>
+
+
+__device__
+void removePrePostWhiteSpaces(char *data, long* start_idx, long* end_idx) {
+	while(data[*start_idx] == ' ' && *start_idx < *end_idx)
+		*start_idx=*start_idx+1;
+	while(data[*end_idx] == ' ' && *start_idx < *end_idx)
+		*end_idx=*end_idx-1;
+}
+
+
+template<typename T>
+__device__
+T convertStrtoInt(char *data, long start_idx, long end_idx) {
```

Comment:
Can you not use an off the shelf function for this?

Reply:
there is not off the shelf code for this, unless we push it to the CPU